core/src/rpc_service.rs

```diff
@@ -358,6 +358,20 @@ impl JsonRpcService {
 
         let ledger_path = ledger_path.to_path_buf();
 
+        // sadly, some parts of our current rpc implemention block the jsonrpc's
+        // _socket-listening_ event loop for too long, due to (blocking) long IO or intesive CPU,
+        // causing no further processing of incoming requests and ultimatily innocent clients timing-out.
+        // So create a (shared) multi-threaded event_loop for jsonrpc and set its .threads() to 1,
+        // so that we avoid the single-threaded event loops from being created automatically by
+        // jsonrpc for threads when .threads(N > 1) is given.
+        let event_loop = {
+            tokio_01::runtime::Builder::new()
+                .core_threads(rpc_threads)
+                .name_prefix("sol-rpc-el")
```

Comment:
btw, I really want clean up our various thread names to conform to the Linux kernel's TASK_COMM_LEN (capped to 15 ascii chars).....

Reply:
Yeah for sure.  I feel the same way.